src/lib.rs

```diff
@@ -145,7 +145,8 @@ impl BitcoinD {
                 let p2p_socket = SocketAddrV4::new(LOCAL_IP, p2p_port);
                 let p2p_arg = format!("-port={}", p2p_port);
                 let connect = format!("-connect={}", other_node_url);
-                let args = vec![p2p_arg, connect];
+                let listen = "-listen=1".to_string(); // With connect specified listen is not defaulted to 1
```

Comment:
If we have always `listen=1` we can't setup network where some nodes are only connected to some other nodes, e.g.
`1 <--> 2 <--> 3` but `1 <-/-> 3`

So perhaps `listen` should be configurable by the user.
And also caller should be able to specify a set of node urls to connect to?

Reply:
Added a boolean to the P2P::Connect variant that basically allow to specify if using listen=1 or listen=0

> And also caller should be able to specify a set of node urls to connect to?

Nodes with `listen=1` should spontaneously connect to each other by their own and one could use `addnode` to facilitate that so I think it's not super necessary specifying a set of nodes